src/MCPServer/tests/test_package.py

```diff
@@ -123,6 +123,107 @@ def test_transfer_get_or_create_from_db_path_with_uuid(tmp_path):
+    new_file = objects_path / "new_file.txt"
+    new_file.touch()
+
+    # One file will be returned from the database  with a UUID, another from
+    # the filesystem without a UUID.
+    for file_count, file_info in enumerate(transfer.files(None, None, "/objects"), 1):
+        assert "%fileUUID%" in file_info
+        assert "%fileGrpUse%" in file_info
+        assert "%relativeLocation%" in file_info
+        if file_info.get("%fileUUID%") != "None":
+            continue
+        assert file_info.get("%relativeLocation%") == str(new_file)
+        file_path = "".join(
+            [
+                transfer.REPLACEMENT_PATH_STRING,
+                "/objects",
+                file_info.get("%relativeLocation%").split("objects")[1],
+            ]
+        )
+        kwargs = {
+            "uuid": uuid.uuid4(),
+            "currentlocation": file_path,
+            "filegrpuse": "original",
+            "transfer_id": transfer_uuid,
+        }
+        models.File.objects.create(**kwargs)
+    assert (
+        file_count == 2
+    ), "Database and file objects were not returned by the generator"
+    assert models.File.objects.filter(transfer_id=str(transfer_uuid)).count() == 2
+
+    # Simulate an additional file object being added later on in the transfer
+    # in a sub directory of the objects folder, e.g. transcribe contents.
+    sub_dir = objects_path / "subdir"
+    sub_dir.mkdir()
+    new_file = sub_dir / "another_new_file.txt"
+    new_file.touch()
+    for file_count, file_info in enumerate(transfer.files(None, None, "/objects"), 1):
+        file_path = "".join(
+            [
+                transfer.REPLACEMENT_PATH_STRING,
+                "/objects",
+                file_info.get("%relativeLocation%").split("objects")[1],
+            ]
+        )
+        if file_info.get("%fileUUID%") != "None":
+            continue
+        kwargs = {
+            "uuid": uuid.uuid4(),
+            "currentlocation": file_path,
+            "filegrpuse": "original",
+            "transfer_id": transfer_uuid,
+        }
+        models.File.objects.create(**kwargs)
+    assert (
+        file_count == 3
+    ), "Database and file objects were not returned by the generator"
+    assert models.File.objects.filter(transfer_id=str(transfer_uuid)).count() == 3
+
+    # Now the database is updated, we will still have the same file count, but
+    # all objects will be returned from the database and we will have uuids.
+    for file_count, file_info in enumerate(transfer.files(None, None, "/objects"), 1):
+        if file_info.get("%fileUUID%") == "None":
+            print(file_info)
+            assert False, "Non-database entries returned from package.files()"
+    assert file_count == 3
+
+    # Clean up state and ensure test doesn't interfere with other transfers
+    # expected to be in the database, e.g. in test_queues.py.
+    models.File.objects.filter(transfer_id=str(transfer_uuid)).delete()
```

Comment:
It doesn't look like it `¯\_(ツ)_/¯` I feel this could be a larger audit task where tests aren't currently being properly isolated. We should play with tox/pytest some more to see what we can get out of it to help do this. 

Providing you're comfortable with this line, I'm happy to keep it. Otherwise, perhaps there's another way to tear down the test I don't know about?

Reply:
I'm OK with the explicit clean up :blush: